src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/Razor/ExtractToCodeBehindCodeActionProviderTest.cs

```diff
@@ -180,6 +180,72 @@ public async Task Handle_InCodeDirective_SupportsFileCreationTrue_ReturnsResult(
         Assert.Equal(42, actionParams.RemoveEnd);
     }
 
+    [Fact]
+    public async Task Handle_AtEndOfCodeDirective_ReturnsResult()
+    {
+        // Arrange
+        var documentPath = "c:/Test.razor";
+        var contents = "@page \"/test\"\n@code { private var x = 1; }";
+        var request = new CodeActionParams()
+        {
+            TextDocument = new TextDocumentIdentifier { Uri = new Uri(documentPath) },
+            Range = new Range(),
+        };
+
+        var location = new SourceLocation(contents.IndexOf(" {", StringComparison.Ordinal), -1, -1);
```

Comment:
Interesting way to get the location, I wonder if we could eventually mark the location with `$$` (like we do in some other tests) to it more obvious where the caret is

Reply:
Yeah, I was just being lazy, copying another test. You're right, this should be better